### DISCUSSION TITLE
Temperature support possible?

### DISCUSSION THREAD

**Discussion by mike-rankin** [HomeSpan/HomeSpan#53] [Ideas]
This is by far the best Homekit library I've ever used. If I could suggest another example sketch, it would be to add and i2c temperature sensor. It would be great to view the data from the sensor on a tile.

**Comment by HomeSpan** [comment]
Thanks for the feedback and suggestion.  Is there a specific temp sensor that uses i2c you recommend?

**Reply by mike-rankin** [reply]
The HDC1080 is a nice little i2c temperature/humidity sensor that I have connected to my ESP32 board now: https://github.com/closedcube/ClosedCube_HDC1080_Arduino
The 12-ServiceLoops example sketch looks very promising to edit. I'll try swapping out the random generated vales for real ones.

**Reply by mike-rankin** [reply]
It appears that it's working. Very messy at the moment but working. In DEV_Sensors.H I've added the line temp->setVal(hdc1080.readTemperature()); and I can see the temperature fluctuate up and down on my iphone when I touch the sensor.
Thanks again for the wonderful library!

**Comment by HomeSpan** [comment]
Glad it's working!  Note that to avoid a lot of network traffic you may want to keep the 5 second timer (or longer) in place.  You could also amend the logic so it includes a timer as well as delta-check.  For example, call setVal() whenever the temperature has changed by at least 1 degree, or every 30 seconds (to refresh any Controllers that may have missed the last update).

**Reply by mike-rankin** [reply]
Agreed. I needed the 5 second timer for testing purposes. If you are interested in adding a hardware temperature/humidity example sketch then I can do the testing on my end or I can ship someone the hardware at no expense.

**Comment by HomeSpan** [comment]
Thanks for the offer though I plan on buying some temp sensors for an outdoor garden monitoring project and can use those as the basis for a new example.   Also, on the main HomeSpan Welcome page I have a link to HomeSpan projects.  It's just a link to https://github.com/topics/homespan so your could add your sketch (assuming it's in GitHub) by simply tagging your project with the word "homespan."
